ios/Capacitor/Capacitor/CapacitorBridge.swift

```diff
@@ -533,6 +533,51 @@ internal class CapacitorBridge: NSObject, CAPBridgeProtocol {
         }
     }
 
+    // MARK: - CAPBridgeProtocol: Paths, Files, Assets
+
+    /**
+     * Translate a URL from the web view into a file URL for native iOS.
+     *
+     * The web view may be handling several different types of URLs:
+     *   - res:// (shortcut scheme to web assets)
+     *   - file:// (fully qualified URL to file on the local device)
+     *   - base64:// (to be implemented)
+     *   - [web view scheme]:// (already converted once to load in the web view, to be implemented)
+     */
+    public func localURL(fromWebURL webURL: URL?) -> URL? {
+        guard let inputURL = webURL else {
+            return nil
+        }
+
+        let url: URL
+
+        switch inputURL.scheme {
+        case "res":
+            guard let resourcePath = Bundle.main.resourcePath else {
+                return nil
+            }
+
+            url = URL(fileURLWithPath: resourcePath).appendingPathComponent("public").appendingPathComponent(inputURL.path)
```

Comment:
To support the embedded case, this should use the configuration's web app location instead of the native app bundle/public path.
```suggestion
            url = config.appLocation.appendingPathComponent(inputURL.path)
```